lib/Sema/CSDiagnostics.h

```diff
@@ -799,20 +799,45 @@ class MissingCallFailure final : public FailureDiagnostic {
   bool diagnoseAsError() override;
 };
 
-class MissingPropertyWrapperUnwrapFailure final : public ContextualFailure {
-  DeclName PropertyName;
+class PropertyWrapperReferenceFailure : public ContextualFailure {
+  DeclName Name;
+  bool IsMemberAccess;
```

Comment:
It seems like this doesn't actually have to be two fields here, you can just use `Optional<DeclName> MemberName` if it's set then it's a member access, otherwise it's contextual mismatch.

Reply:
Also I'd always try to make it clear everywhere and avoid using `Name` unless it could only have one meaning, otherwise in this case it wasn't clear to me what name is this - property or member referred through the property?